ingest/lambdas/metadata-downloader.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+import * as dspaceClient from '../../lib/dspace-client';
+import { getStringFromEnv } from '../../lib/env-utils';
+import { putJson, S3ObjectLocation } from '../../lib/s3-utils';
+
+const snsEventSchema = z.object({
+  Records: z.array(
+    z.object({
+      Sns: z.object({
+        Message: z.string().uuid(),
+      }),
+    })
+  ).nonempty(),
+});
+
+export const handler = async (event: unknown) => {
+  const snsEvent = snsEventSchema.parse(event);
```

Comment:
I really like this pattern. It feels broken that every lambda we write doesn't start like this. 👍